src/app/compare/compare-tree/compare-tree.component.ts

```diff
@@ -1,209 +1,156 @@
-    this.rightReport = this.rightTreeReference.getItems()[0];
-    this.leftTreeReference.selectItem(this.leftReport);
-    this.rightTreeReference.selectItem(this.rightReport);
-    this.compareEvent.emit({
-      leftReport: this.leftReport,
-      rightReport: this.rightReport,
-    });
-  }
-
-  makeLabelsRed(left: any, right: any) {
-    this.redLabel(left);
-    this.redLabel(right);
-  }
-
-  redLabel(item: any) {
-    item.label = `<span style='color: red;'>${item.label}</span>`;
   }
 
-  iterateToMakeLabelsRed(leftItem: any, rightItem: any) {
-    let result = this.checkIfLabelsDifferent(leftItem, rightItem);
-    const shortestTreeLength = Math.min(leftItem.items.length, rightItem.items.length);
-    this.makeRestOfTreesRed(shortestTreeLength, rightItem.items, leftItem.items);
-
-    for (let i = 0; i < shortestTreeLength; i++) {
-      let both = this.iterateToMakeLabelsRed(leftItem.items[i], rightItem.items[i]);
-      leftItem.items[i] = both.left;
-      rightItem.items[i] = both.right;
+  selectCheckpoint(tree: NgSimpleFileTree, item: FileTreeItem, checkpointToMatch: Checkpoint): void {
+    if (item.children) {
+      for (const child of item.children) {
+        const checkpoint = child.originalValue as Checkpoint;
+        if (this.getCheckpointId(checkpoint.uid) === this.getCheckpointId(checkpointToMatch.uid)) {
+          tree.selectItem(child.path);
+          return;
+        } else {
+          this.selectCheckpoint(tree, child, checkpointToMatch);
+        }
+      }
     }
-
-    return result;
   }
 
-  makeRestOfTreesRed(startPoint: number, leftItems: any[], rightItems: any[]) {
-    let items = leftItems;
-    if (rightItems.length > leftItems.length) items = rightItems;
-    for (let i = startPoint; i < items.length; i++) {
-      this.redLabel(items[i]);
+  namesMatch(name1: string | undefined, name2?: string | undefined): string {
+    if (name1 && name2 && name1 === name2) {
+      return '';
     }
-  }
-
-  checkIfLabelsDifferent(left: any, right: any): { left: any; right: any } {
-    if (left.level > -1) {
-      if (left.value.message !== right.value.message) this.makeLabelsRed(left, right);
-    } else {
-      if (left.value.xml !== right.value.xml) this.makeLabelsRed(left, right);
+    return 'red';
+  }
+
+  findCorrespondingCheckpoint(
+    itemToMatch: Checkpoint | Report,
+    report: Report | Checkpoint | undefined,
+  ): Checkpoint | undefined {
+    if (report) {
+      const checkpoints: Checkpoint[] = report.checkpoints ?? [];
+      for (let checkpoint of checkpoints) {
+        if (
+          ReportUtil.isCheckPoint(itemToMatch) &&
+          this.getCheckpointId(checkpoint.uid) === this.getCheckpointId(itemToMatch.uid)
+        ) {
+          return checkpoint;
+        }
+        if (checkpoint.checkpoints) {
+          return this.findCorrespondingCheckpoint(itemToMatch, checkpoint);
+        }
+      }
     }
+    return undefined;
```

Comment:
I'm not sure I like undefined here, is there a reason it's undefined instead of null?
For context, I see undefined as something not defined in memory yet while null is when you can't return the actual object for some reason.
If this function is used to assign a variable (`const x`), having a null value would make more sense then having `const x` still returning undefined even after assignment